src/java/org/apache/cassandra/repair/RepairRunnable.java

```diff
@@ -131,167 +145,239 @@ protected void fireProgressEvent(ProgressEvent event)
         }
     }
 
-    protected void fireErrorAndComplete(int progressCount, int totalProgress, String message)
+    private void skip(String msg)
+    {
+        logger.info("Repair {} skipped: {}", parentSession, msg);
+        fireProgressEvent(new ProgressEvent(ProgressEventType.NOTIFICATION, 100, 100, "Repair " + parentSession + " skipped: " + msg));
```

Comment:
When there is no valid CFs to be repaired, a skip event with 100/100 progress is fired, and then another `success` event with `progressCounter/totalProgress` is fired in `success()`.. 

Should we use `progressCounter/totalProgress` for skip progress event?

Reply:
switched to that and refactored all usage to use a new `notification` method.